### PR TITLE
Refactor api calls

### PR DESCRIPTION
**Pull request by nicovigil1**
adds a github service model that handles api calls - also adds a UserRelations for followers and following & refactors the controller.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,11 @@
 class UsersController < ApplicationController
   def show
     if current_user.token
-      @repos = Repo.generate(conn(current_user.token)).take(5)
+      raw_repos = GithubService.repos(current_user)
```
Comment (danbriechle) [inline app/controllers/users_controller.rb]: Lets get rid of this raw repos

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -20,13 +24,13 @@ def create
...
   end
 
-  def conn(token)
-    conn = Faraday.new(url: "https://api.github.com") do |f|
-      f.headers["Authorization"] = "Token #{token}"
-      f.adapter Faraday.default_adapter
-    end
-    conn.get "/user/repos"
-  end
+  # def conn(token)
```
Comment (danbriechle) [inline app/controllers/users_controller.rb]: lets get rid of this commented out stuff

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (nicovigil1) [reply app/controllers/users_controller.rb]: yeah, good catch - I completely missed that. We should be able to get it when we merge - I suggest we do so locally?

File: app/models/github.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "rails_helper"
```
Comment (danbriechle) [inline app/models/github.rb]: lets get rid of all f this

Comment (nicovigil1) [reply app/models/github.rb]: oops. Completely forgot to delete this - that's what I get for coding over lunch

File: app/models/repo.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@ class Repo
   def initialize(gh_response)
     @url = gh_response["html_url"]
     @name = gh_response["name"]
+    @gh_response = gh_response
```
Comment (danbriechle) [inline app/models/repo.rb]: we can get rid of this guy too

Comment (nicovigil1) [reply app/models/repo.rb]: yup. leftover from when we had less of an idea of how to work with json.

File: app/models/user_relation.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    attr_reader :url, :username
+    def initialize(aspects)
+        @url = aspects["html_url"]
+        @username = aspects["login"]
+    end 
+
+    def self.generate(raw_data)
+        parsed_data = JSON.parse(raw_data.body)
+        parsed_data.map do |relation|
+            test = UserRelation.new(relation)
```
Comment (danbriechle) [inline app/models/user_relation.rb]: we could probably get rid of the test =

Comment (nicovigil1) [reply app/models/user_relation.rb]: All these tiny oversights are embarrassing

**Comment by nicovigil1** [comment]
Alright we'll make these changes when we merge locally!

**Pull request closed without merging**